src/main/java/com/minecolonies/coremod/colony/jobs/JobDeliveryman.java

```diff
@@ -87,61 +131,104 @@ public SoundEvent getMoveAwaySound()
         return null;
     }
 
-    @Override
-    public SoundEvent getBedTimeSound()
+    /**
+     * Returns whether or not the job has a currentTask.
+     *
+     * @return true if has currentTask, otherwise false.
+     */
+    public boolean hasTask()
     {
-        if (getCitizen() != null)
+        return !taskQueue.isEmpty() || returning;
+    }
+
+    /**
+     * Returns the {@link IRequest} of the current Task.
+     *
+     * @return {@link IRequest} of the current Task.
+     */
+    public IRequest<? extends Delivery> getCurrentTask()
+    {
+        if (taskQueue.isEmpty())
         {
-            return getCitizen().isFemale() ? DeliverymanSounds.Female.offToBed : null;
+            return null;
         }
-        return null;
+        return getColony().getRequestManager().getRequestForToken(taskQueue.peekFirst());
     }
 
     /**
-     * Generate your AI class to register.
+     * Method used to add a request to the queue
      *
-     * @return your personal AI instance.
+     * @param token The token of the requests to add.
      */
-    @NotNull
-    @Override
-    public AbstractAISkeleton<JobDeliveryman> generateAI()
+    public void addRequest(@NotNull final IToken token)
     {
-        return new EntityAIWorkDeliveryman(this);
+        taskQueue.add(token);
     }
 
     /**
-     * Returns whether or not the job has a destination.
+     * Method called to mark the current request as finished.
      *
-     * @return true if has destination, otherwise false.
+     * @param successful True when the processing was successful, false when not.
      */
-    public boolean hasDestination()
+    public void finishRequest(@NotNull final boolean successful)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "successful" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)